### PR TITLE
Add Codecov test coverage

### PR DESCRIPTION
**Pull request by ehmicky**
This adds Codecov for test coverage.

It's quite useful to see the test coverage of a PR by looking at the PR status. 
This disables PR comments because they can be noisy.

### CODE REVIEW

File: scripts/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+os="$1"
+os="${os/-latest/}"
+
+node="$2"
+node="node_${node//./_}"
+
+# We don't use the -Z flag (which makes CI build fail on upload error) because
+# Codecov fails wait too often.
+curl -s https://codecov.io/bash | \
+  bash -s -- -f coverage/coverage-final.json -F "$os" -F "$node"
```
Comment (ehmicky) [inline scripts/coverage.sh:13]: We need a Bash script in order to pass the `-F` tags (they do not accept `.` nor `-`).

**Review by erezrokah — CHANGES_REQUESTED** [review]
Cool 😄 Looks like it is failing at the moment:
https://github.com/netlify/cli/pull/1440/checks?check_run_id=1266323266#step:7:33

**Comment by ehmicky** [comment]
This is a [known bug](https://github.com/codecov/codecov-bash/issues/330) in Codecov.
Apparently, the Codecov report are still looking good, so I am assuming the data is still uploaded. See report [here](https://codecov.io/gh/netlify/cli/pull/1440/tree?path=src). From that angle, would it still be useful to merge?

**Comment by erezrokah** [comment]
> would it still be useful to merge?

Sure, go for it

**Review by erezrokah — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -36,3 +36,5 @@ jobs:
           NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
           IS_FORK: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == true }}
           NETLIFY_TEST_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Codecov test coverage
```
Comment (erezrokah) [inline .github/workflows/main.yml:39]: Also, should be do this just for one node version and OS? Any benefit from uploading all results? Does Codecov knows to differentiate?

Comment (ehmicky) [reply .github/workflows/main.yml:39]: Yes, it merges them together, and you can even see per-OS results.
That's useful for OS-specific code. For example, if some code only runs on Windows, and another never runs on Windows, we need to run both to get the proper test coverage.

**Pull request merged**